lua-mode.el

```diff
@@ -304,7 +315,7 @@ If the latter is nil, the keymap translates into `lua-mode-map' verbatim.")
  key like `{' is pressed")
 (make-variable-buffer-local 'lua-electric-flag)
 
-(defcustom lua-prompt-regexp "[^\n]*\\(>[\t ]+\\)+$"
+(defcustom lua-prompt-regexp "^\\(>[\t ]+\\)+"
```

Comment:
Will this regexp break for code that does `io.write` and doesn't append a newline? E.g.
```
$ lua
Lua 5.2.4  Copyright (C) 1994-2015 Lua.org, PUC-Rio
> io.write('asdf')
asdf> 
```

Reply:
I hadn't thought of that.  Interestingly, for me I get:

```lua
Lua 5.3.5  Copyright (C) 1994-2018 Lua.org, PUC-Rio
No entry for terminal type "emacs";
using dumb terminal settings.
> io.write('asdf')
asdffile (0x7fff9d8f21a8)
> 
```

But the old version will fail for prompts with more information after them, as in the Shell itself.  We have to tie either to the bol or eol; most comint-major modes with shells choose the former.